Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -11,10 +11,135 @@
 #include <AMReX_MultiFab.H>
 
 
+/**
+* \brief Get the value of arr in the neighbor (i_n, j_n) on the plane with normal 'dim'.
+*        I.E. If dim==0 it return arr(i, j + i_n, k + j_n),
+*             if dim==1 it return arr(i + i_n, j, k + j_n),
+*             if dim==2 it return arr(i + i_n, j + j_n, k)
+*
+* \param[in] arr \c Array4 To be accessed
+* \param[in] i, j, k the indices of the "center" cell
+* \param[in] i_n the offset of the neighbor in the first direction
+* \param[in] j_n the offset of the neighbor in the second direction
+* \param[in] dim normal direction to the plane in consideration (0 for x, 1 for y, 2 for z)
+*/
+template <class T>
+AMREX_GPU_DEVICE
+constexpr
+T
+GetNeigh(const amrex::Array4<T>& arr,
```

Comment:
Since this is read-only, we should be able to do:
```suggestion
GetNeigh(const amrex::Array4<const T>& arr,
```

Reply:
Wow I didn't know I could do this 🤯

Reply:
Unfortunately, it doesn't seem to work.
Whenever I call GetNeigh afterwards I get this error

`
No matching function for call to 'GetNeigh' candidate template ignored: cannot deduce a type for 'T' that would make 'const T' equal 'double'
`

Reply:
Ok, that's likely because the array we pass cannot be implicitly casted from `Array4<double>` to `Array4<const double>`.